rllib/algorithms/algorithm.py

```diff
@@ -1846,8 +1854,23 @@ def add_policy(
             policy_state=policy_state,
             policy_mapping_fn=policy_mapping_fn,
             policies_to_train=policies_to_train,
+            module_spec=module_spec,
         )
 
+        # If learner API is enabled, we need to also add the underlying module
+        # to the learner group.
+        if self.config._enable_learner_api:
+            policy = self.get_policy(policy_id)
+            module = policy.model
+            self.learner_group.add_module(
+                module_id=policy_id,
+                module_spec=SingleAgentRLModuleSpec.from_module(module),
+            )
+
+            # TODO (Kourosh): Only allow users to set weights of a particular module
```

Comment:
This is one candidate where we could optimize to speed up weight adding a module

Reply:
done